cloudsync/provider.py

```diff
@@ -408,19 +408,24 @@ def split(self, path):
             return self.sep, path[index + 1:]
         return path[:index], path[index+1:]
 
-    def normalize_path(self, path: str):
-        """Used internally for comparing paths in a case and sep insensitive manner, as appropriate."""
-        norm_path = path.rstrip(self.sep)
-        if self.sep in ["\\", "/"]:
-            parts = re.split(r'[\\/]+', norm_path)
-        else:
-            parts = re.split(r'[%s]+' % self.sep, norm_path)
+    def normalize_path(self, path: str, pretty_leaf: bool = False):
+        """Used internally for comparing paths in a case and sep insensitive manner, as appropriate.
+
+        Args:
+            path: the path to normalize
+            pretty_leaf: when true, preserve case of path's leaf node
+        """
+        if self.alt_sep:
+            path = path.replace(self.alt_sep, self.sep)
+        sep_regexp = r"[\\]+" if self.sep == "\\" else f"[{self.sep}]+"
+        parts = re.split(sep_regexp, path.rstrip(self.sep))
         norm_path = self.join(*parts)
-        if not self.case_sensitive:
-            norm_path = norm_path.lower()
-        else:
-            norm_path = self.join(self.dirname(norm_path).lower(), self.basename(norm_path))
-        return norm_path
+
+        if self.case_sensitive:
+            return norm_path
+        elif pretty_leaf:
```

Comment:
Lint complains about putting an else after a return

Reply:
lint is passing with this as is, and we have a few places in this file where we do the same thing. can change this if anyone has a strong opinion about this, though.

Reply:
i like else statements better